### PR TITLE
IBP-6185 build site admin with angular 7v

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

Include:
https://github.com/IntegratedBreedingPlatform/BMSAPI/pull/802
https://github.com/IntegratedBreedingPlatform/Workbench/pull/819

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/api/user/UserSearchRequest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		this.status = status;
+	}
+
+	public Integer getRoleId() {
+		return this.roleId;
+	}
+
+	public void setRoleId(final Integer roleId) {
+		this.roleId = roleId;
+	}
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/user/UserSearchRequest.java:26]: @cuenyad add equals, hashCode, etc..

File: src/main/java/org/generationcp/middleware/dao/RoleDAO.java [hunk 2]
```diff
@@ -58,4 +61,32 @@ public Role getRoleById(final Integer id) {
 		return role;
 	}
 
+	public long countRolesUsers(final RoleSearchDto roleSearchDto) {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/dao/RoleDAO.java:64]: @cuenyad this name should be countRoles only? I dont see anything related to users here

File: src/main/java/org/generationcp/middleware/dao/RoleDAO.java [hunk 3]
```diff
@@ -58,4 +61,32 @@ public Role getRoleById(final Integer id) {
...
+					criteria.add(Restrictions.eq("roleType.id", roleSearchDto.getRoleTypeId()));
+				}
+				if (StringUtils.isNotBlank(roleSearchDto.getName())) {
+					criteria.add(Restrictions.eq("name", roleSearchDto.getName()));
+				}
+			}
+			return criteria.list().size();
+
+		} catch (final HibernateException e) {
+			final String message = "Error with countRolesUsers query from RoleDAO: " + e.getMessage();
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/dao/RoleDAO.java:86]: rename to countRoles here

File: src/main/java/org/generationcp/middleware/dao/RoleDAO.java (hunk 2 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/dao/RoleDAO.java]: done!

File: src/main/java/org/generationcp/middleware/dao/RoleDAO.java (hunk 3 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/dao/RoleDAO.java]: done!

File: src/main/java/org/generationcp/middleware/api/user/UserSearchRequest.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/user/UserSearchRequest.java:26]: Done!

**Pull request merged**